core/include/scipp/core/transform.h

```diff
@@ -273,18 +273,16 @@ struct TransformSparse {
 };
 
 template <class Op, class... Args>
-constexpr static bool check_all_or_none_variances(const Op &,
-                                                  const Args &... args) {
+constexpr static bool
+check_all_or_none_variances(const Op &, const Args &... valAndVariances) {
   constexpr bool force_same =
       std::is_base_of_v<transform_flags::expect_all_or_none_have_variance_t,
                         Op>;
-  constexpr bool all_variances =
-      force_same &&
-      (is_ValuesAndVariances_v<std::decay_t<decltype(args)>> && ...);
-  constexpr bool variances =
-      all_variances ||
-      (is_ValuesAndVariances_v<std::decay_t<decltype(args)>> || ...);
-  return force_same && !all_variances && variances;
+  return force_same && !(force_same && (valAndVariances && ...)) &&
+         ((force_same && (valAndVariances && ...)) || (valAndVariances || ...));
```

Comment:
```suggestion
         (valAndVariances || ...);
```
should be equivalent.